internal/executor/build/task.go

```diff
@@ -0,0 +1,90 @@
+package build
+
+import (
+	"fmt"
+	"github.com/cirruslabs/cirrus-ci-agent/api"
+	"github.com/cirruslabs/cirrus-cli/internal/executor/build/taskstatus"
+	"github.com/cirruslabs/cirrus-cli/internal/executor/instance"
+	"strconv"
+	"sync"
+	"time"
+)
+
+const defaultTaskTimeout = 60 * time.Minute
+
+type Task struct {
+	ID       int64
+	status   taskstatus.Status
+	Instance *instance.Instance
+	Timeout  time.Duration
+
+	// Original Protocol Buffers structure for reference
+	ProtoTask *api.Task
+
+	// A mutex to guarantee safe accesses from both the main loop and gRPC server handlers
+	Mutex sync.Mutex
```

Comment:
If we are going the mutex way let's use `RWMutex` here.

Reply:
See dce79d79cd0a1469b4605d36e2fabb8d64703f71.